### ISSUE TITLE
null exception

### ISSUE THREAD

**Issue by joyola** [myflashlab/alarmManager-ANE#5]
**Application-app.xml**

```
<manifest android:installLocation="auto">
	<uses-sdk android:minSdkVersion="9" android:targetSdkVersion="22" />

    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
    <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.GET_ACCOUNTS"/>
	<uses-permission android:name="android.permission.WAKE_LOCK" />
	<uses-permission android:name="android.permission.VIBRATE"/>
	<uses-permission android:name="com.android.alarm.permission.SET_ALARM" />
	<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />

	<permission android:name="air.com.yourcompany.yourgame.permission.C2D_MESSAGE" android:protectionLevel="signature" />
	<uses-permission android:name="air.com.dev4evolution.permission.C2D_MESSAGE" />
	<uses-permission android:name="com.google.android.c2dm.permission.RECEIVE" />

	<application>
		<activity android:excludeFromRecents="false">
			<intent-filter>
				<action android:name="android.intent.action.MAIN"/>
				<category android:name="android.intent.category.LAUNCHER"/>
			</intent-filter>
			<intent-filter>
				<action android:name="android.intent.action.VIEW" />
				<category android:name="android.intent.category.BROWSABLE" />
				<category android:name="android.intent.category.DEFAULT" />
				<data android:scheme="air.com.doitflash.exAlarm" />
			</intent-filter>
		</activity>

		<!-- manual or onesignal mode -->
		<receiver android:name="com.milkmangames.extensions.android.push.GCMBroadcastReceiver" android:permission="com.google.android.c2dm.permission.SEND" >
		  <intent-filter>
			<action android:name="com.google.android.c2dm.intent.RECEIVE" />
			<action android:name="com.google.android.c2dm.intent.REGISTRATION" />
			<category android:name="air.com.dev4evolution" />
		  </intent-filter>
		</receiver>						

		<service android:name="com.milkmangames.extensions.android.push.GCMIntentService" />
		
		<!-- alarm task -->
		<receiver android:name="com.doitflash.alarmManager.alarmTypes.Alarm_ShowToast" />
		<receiver android:name="com.doitflash.alarmManager.alarmTypes.Alarm_Notification" />
		<receiver android:name="com.doitflash.alarmManager.alarmTypes.Alarm_SimpleDispatch" />

		<receiver android:name="com.doitflash.alarmManager.RebootAutoStartAlarms">
			<intent-filter>
				<action android:name="android.intent.action.BOOT_COMPLETED" />
			</intent-filter>
		</receiver>
		
			<activity android:name="com.doitflash.alarmManager.WakeActivity" /> 
	</application>
</manifest>
```

**Main.xmlx**

```
NativeApplication.nativeApplication.addEventListener(Event.ACTIVATE, handleActivate);
NativeApplication.nativeApplication.addEventListener(Event.DEACTIVATE, handleDeactivate);
NativeApplication.nativeApplication.addEventListener(InvokeEvent.INVOKE, onInvoke);

var alarm:Alarm = new Alarm();
alarm.addEventListener(AlarmEvent.ALARM_SIMPLE_DISPATCH, alarmSimpleDispatch);

var alarmTask:Alarm_SimpleDispatch = new Alarm_SimpleDispatch();
alarmTask.setWakeUp(false);
alarmTask.setDelay(1000);
alarmTask.setMassage("this is a msg!");
				
alarm.setAlarm(alarmTask.data);

```

I get this exception:

TypeError: Error #1009: Cannot access a property or method of a null object reference.
	at com.myflashlab.air.extensions.timerTask.android::Android()
	at com.myflashlab.air.extensions.timerTask::Alarm()
	at views::MainHomeView/onInitializeApplicationTaskGroupComplete()[/Users/joyola/Documents/Adobe Flash Builder 4.6/salesforce-mobile/src/views/MainHomeView.mxml:81]
	at flash.events::EventDispatcher/dispatchEventFunction()
	at flash.events::EventDispatcher/dispatchEvent()
	at org.spicefactory.lib.task::Task/fireCompleteEvent()[/Users/jenshalm/Projects/Spicefactory3/Legacy-Spicelib-Tasks/src/main/org/spicefactory/lib/task/Task.as:537]
	at org.spicefactory.lib.task::Task/complete()[/Users/jenshalm/Projects/Spicefactory3/Legacy-Spicelib-Tasks/src/main/org/spicefactory/lib/task/Task.as:382]
	at org.spicefactory.lib.task::SequentialTaskGroup/nextTask()[/Users/jenshalm/Projects/Spicefactory3/Legacy-Spicelib-Tasks/src/main/org/spicefactory/lib/task/SequentialTaskGroup.as:95]
	at org.spicefactory.lib.task::SequentialTaskGroup/handleTaskComplete()[/Users/jenshalm/Projects/Spicefactory3/Legacy-Spicelib-Tasks/src/main/org/spicefactory/lib/task/SequentialTaskGroup.as:85]
	at org.spicefactory.lib.task::TaskGroup/completeChild()[/Users/jenshalm/Projects/Spicefactory3/Legacy-Spicelib-Tasks/src/main/org/spicefactory/lib/task/TaskGroup.as:320]
	at org.spicefactory.lib.task::Task/complete()[/Users/jenshalm/Projects/Spicefactory3/Legacy-Spicelib-Tasks/src/main/org/spicefactory/lib/task/Task.as:380]
	at util.tasks::GettingPedidosTask/onCompleteGettingPedidos()[/Users/joyola/Documents/Adobe Flash Builder 4.6/salesforce-mobile/src/util/tasks/GettingPedidosTask.as:48]
	at Function/http://adobe.com/AS3/2006/builtin::call()
	at util::SqlResultArrayCollectionResponder/onResult()[/Users/joyola/Documents/Adobe Flash Builder 4.6/salesforce-mobile/src/util/SqlResultArrayCollectionResponder.as:54]

**Comment by myflashlab** [comment]
We noticed that you are missing the ```Theme.Transparent``` in the following activity: ```<activity android:name="com.doitflash.alarmManager.WakeActivity" android:theme="@style/Theme.Transparent" />```

However, I'm not sure if the error code could be related this.

Are you sure you are running your app on a real device? because simulators cannot run this ANE.

**Comment by joyola** [comment]
I remove the theme tag because an error occurr when I run the application in Samsung Tab 4.

![image](https://cloud.githubusercontent.com/assets/4267418/20524119/99dcacfc-b097-11e6-8af9-183b397aa7c4.png)

**Comment by myflashlab** [comment]
Here are some of the things you could look into for troubleshooting the problem:

1. Always use the latest AIR SDK.
2. Try to use the ANE in a simple project without using other ANEs to see how it goes especially not using it with other ANEs from other providers because we cannot know how they are managing things on their ANEs so it would be very probable that ANEs are conflicting with each other.
3. Run the ANE on a real device not the simulator.

**Comment by myflashlab** [comment]
we have just launched a new ANE, i.e [Local Notifications](https://github.com/myflashlab/LocalNotification-ANE) which works on Android and iOS. Considering that the alarm manager ANE was only used for the its local notification ability, we decided to cut down the confusion and re-build a brand new ANE. I'm leaving you this note as we not longer maintain the alarm manager ANE and instead will support LocalNotification. If you have purchased the Alarm ANE already, drop us a note [here](http://www.myflashlabs.com/contact/) and my colleagues will give you access to the replaced ANE for free. Thanks.
